### PR TITLE
Remove unused components, improve tests, clean up.

### PR DESCRIPTION
**Pull request by ensary**
* Removed templates and commands that support them. The templates are out of date, and we no longer use them
* Added tests via pytest+testinfra (WIP, we need more)
* Switched to compose coming from the distro instead of PIP

Next steps - upgrade to debian bullseye, more tests, extract python deps and go deps to `requirements.txt` and `go.mod`, this way we can get updates via renovate.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -53,21 +50,10 @@ RUN curl -sfL https://deb.nodesource.com/setup_12.x | bash - && \
 
 FROM js_deps AS python_deps
 
-RUN apt-get install -y locales python3-distutils
-RUN curl https://bootstrap.pypa.io/get-pip.py | python3
-RUN pip3 install -U setuptools cookiecutter
+RUN apt-get install -y locales python3-distutils python3-pip
```
Comment (dkolbly) [inline Dockerfile:53]: Why do we use python here?

**Review by dkolbly — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (ensary) [reply Dockerfile:53]: The `sdcli` is supposed to support golang + python + node projects.

Comment (mikerott) [reply Dockerfile:53]: > supposed to support
ummm, sort of.  The original intent was python linting, but I don't think we did anything more than that, including anything for js.  Maybe worth putting common lint/format/etc rules in for multiple languages (java *cough)

**Review by mikerott — APPROVED** [review]
Approved, with a few minor questions:

* While we're here deleting things, shall we delete codecov.yml?
* You'll git tag with a major version change, since this is a breaking change?

File: Dockerfile (hunk 1 above)
Comment (ensary) [reply Dockerfile:53]: Let's discuss. I am def for chopping up sdcli into more targeted parts/

**Comment by ensary** [comment]
@mikerott the intent is to tag this v3 (we had v2 at some point before) once I add things that will allow renovate to auto-update reqs (requirements.txt + go.mod for go tools). 
Two more planned changes - newer debian version (the one we are using is close to EOL apart from LTS support) and newer node version (we are using EOLed one according to the installer).

**Comment by mikerott** [comment]
Ok so this PR commit level doesn't get tagged.  Got it.

**Pull request merged**